front-end/src/component/Dashboard.js

```diff
@@ -1,7 +1,42 @@
-import { React } from 'react'
+import { React, useEffect } from 'react'
 import { Link } from 'react-router-dom'
+import endPoints from '../config/fetch.js'
 
 const Dashboard = () => {
+    const userInformation = {
+        username: null,
+        email: null,
+        id: null
+    };
+
+    useEffect(() => { /*Needs test case by front-end team*/
+        const findUserInfo = async () => {
+            try {
+                const response = await fetch(endPoints.loginEndpoint, {
+                    method: 'GET',
+                    headers: {
+                        'Content-Type': 'application/json', // Set the content type to JSON
+                    },
+                    credentials: 'include' // Include credentials in the request to allow browser to send cookis
+                });
+
+                if (response.ok) {
+                    const userData = await response.json();//resolves promis and turns the fetch information into JS object 
+                    userInformation.username = userData.username;
+                    userInformation.email = userData.email;
+                    userInformation.id = userData.id;
+                };
+            }
+            catch (error) {
+                return;//we dont want to console log a error here, because its just checking if the user logged in or not
+            }
+        };
+
+        findUserInfo();
```

Comment:
why the nested function in the useEffect? Cant we just have one anonymous asynchronous function as the input for the useEffect that just contains the try/catch block? Instead of passing a function that create and calls another function?

Reply:
No, use effect first function, from testing, cannot be async so we just need a function within. This is common coding style in JS, functions inside functions.

Reply:
Gotcha no problem